### PR TITLE
Delete TNF build artifacts

### PR DESCRIPTION
**Pull request by betoredhat**
- Delete dangling containers and images used to build TNF

### CODE REVIEW

File: roles/cnf_cert/tasks/teardown.yml [hunk 1]
```diff
@@ -1,5 +1,17 @@
 ---
 # Deleting resources created by the CNF Cert Suite to support the execution
+- name: Prune Buildah images and containers
```
Comment (ramperher) [inline roles/cnf_cert/tasks/teardown.yml:3]: Note the teardown tasks are not executed in the DCI teardown stage, they're executed just after cnf_cert/tests.yml. If the job fails in the tests.yml, then these tasks will not be executed. That's why the "Remove local tnf image" task was just after the execution of tnf.

I think there are two options here:

- Put again the image removal tasks after tnf execution in tests.yml
- Think on a more sophisticated way of implementing this, e.g. call teardown.yml in DCI teardown stage and include the logic in main.yml to select what to do (maybe this is the most interesting one)

Comment (betoredhat) [reply roles/cnf_cert/tasks/teardown.yml]: Thanks @ramperher. I want to hold on a bit on running the buildah cleanup as it is going to remove the dangling artifacts even those that were not created by the CNF image build. May be better to run it from our cleanup scripts.  Let's see if the  `--force-rm` during the image build helps a bit. 

Cheers

File: roles/cnf_cert/tasks/pre-run.yml [hunk 2]
```diff
@@ -111,6 +111,7 @@
       ansible.builtin.shell: |
         set -x
         podman build -t {{ tnf_image }} \
+        --force-rm \
```
Comment (thekad) [inline roles/cnf_cert/tasks/pre-run.yml:114]: are you sure this does what you think it does? this flag defaults to true

Comment (thekad) [reply roles/cnf_cert/tasks/pre-run.yml:114]:
```
   --force-rm
       Always remove intermediate containers after a build, even if the  build  fails  (de‐
       fault true).
```

Comment (betoredhat) [reply roles/cnf_cert/tasks/pre-run.yml:114]: well, I didn't realize about the default. Will close this. Thanks

**Pull request closed without merging**